src/ssl.c

```diff
@@ -48914,6 +49416,99 @@ int wolfSSL_BN_sub(WOLFSSL_BIGNUM* r, const WOLFSSL_BIGNUM* a,
     return 0;
 }
 
+WOLFSSL_API int wolfSSL_BN_mul(WOLFSSL_BIGNUM *r, WOLFSSL_BIGNUM *a, WOLFSSL_BIGNUM *b,
+    WOLFSSL_BN_CTX *ctx)
+{
+    int ret = WOLFSSL_SUCCESS;
+
+    (void)ctx;
+
+    WOLFSSL_ENTER("wolfSSL_BN_mul");
+
+    if (r == NULL || a == NULL || b == NULL || r->internal == NULL ||
+        a->internal == NULL || b->internal == NULL) {
+        ret = WOLFSSL_FAILURE;
+    }
+
+    if (ret == WOLFSSL_SUCCESS) {
+        ret = mp_mul((mp_int*)a->internal, (mp_int*)b->internal,
+                     (mp_int*)r->internal);
+        if (ret == MP_OKAY) {
+            ret = WOLFSSL_SUCCESS;
+        }
+        else {
+            ret = WOLFSSL_FAILURE;
+        }
+    }
+
+    WOLFSSL_LEAVE("wolfSSL_BN_mul", ret);
+
+    return ret;
+}
+
+int wolfSSL_BN_div(WOLFSSL_BIGNUM* dv, WOLFSSL_BIGNUM* rem,
+                   const WOLFSSL_BIGNUM* a, const WOLFSSL_BIGNUM* d,
+                   WOLFSSL_BN_CTX* ctx)
+{
+    int ret = WOLFSSL_SUCCESS;
+
+    (void)ctx;
+
+    WOLFSSL_ENTER("wolfSSL_BN_div");
+
+    if (dv == NULL || rem == NULL || a == NULL || d == NULL ||
+        dv->internal == NULL || rem->internal == NULL || a->internal == NULL ||
+        d->internal == NULL) {
+        ret = WOLFSSL_FAILURE;
+    }
+
+    if (ret == WOLFSSL_SUCCESS) {
+        ret = mp_div((mp_int*)a->internal, (mp_int*)d->internal,
+                     (mp_int*)dv->internal, (mp_int*)rem->internal);
+        if (ret == MP_OKAY) {
+            ret = WOLFSSL_SUCCESS;
+        }
+        else {
+            ret = WOLFSSL_FAILURE;
+        }
+    }
+
+    WOLFSSL_LEAVE("wolfSSL_BN_div", ret);
+
+    return ret;
+}
+
+#ifdef WOLFSSL_KEY_GEN /* Needed to get mp_gcd. */
+int wolfSSL_BN_gcd(WOLFSSL_BIGNUM* r, WOLFSSL_BIGNUM* a, WOLFSSL_BIGNUM* b,
+                   WOLFSSL_BN_CTX* ctx)
+{
+    int ret = WOLFSSL_SUCCESS;
+
+    (void)ctx;
+
+    WOLFSSL_ENTER("wolfSSL_BN_gcd");
+
+    if (r == NULL || a == NULL || b == NULL) {
```

Comment:
You'll need to add `->internal == NULL` checks here too. Looks like most of the mp_ API's do not do NULL checks.